### DISCUSSION TITLE
does not find some users

### DISCUSSION THREAD

**Discussion by l1we4** [Rapptz/discord.py#5856] [Q&A]
## Summary

I use convert and fetch_user(id) programs and user non found

## Reproduction Steps
            ```idd = user or str(ctx.author)
            try:
                member = await commands.UserConverter().convert(ctx, idd)
            except:
                member = await self.bot.fetch_user(idd)
            else:
                member = discord.utils.get(ctx.author.guild.members, name = member.name)
            print(member)```

## Expected Results

Get member or info about him

## Actual Results

(None)

## System Information

-> Version Discord.py (1.4.1)
-> Version Python (3.7.9)
-> Windows 10 x64

**Comment by ChrisGallardo** [comment]
Hey @l1we4, `fetch_user` takes an `int` but you are passing a `bool`. I think you aren't using the `or` operator correctly. `or` is `True` if either statements are `True`. So in your code, `idd = user or str(ctx.author)`, `user` and `str(ctx.author)` are defined, making them `True`. `idd` is going to equal `True`.

**Reply by AbstractUmbra** [reply]
Sadly this is not the case:

```py
In [1]: val_a = None                                                                                                                                                                                                                                                      

In [2]: val_b = 12                                                                                                                                                                                                                                                        

In [3]: val_c = val_a or val_b                                                                                                                                                                                                                                            

In [4]: val_c                                                                                                                                                                                                                                                             
Out[4]: 12
```

The `or` operator here sets `idd` to reference `user` OR `str(ctx.author)`, whichever is not False/falsey first.

**Comment by AbstractUmbra** [comment]
The reason this is failing is the `UserConverter.convert` method will be failing, but your bare `except` will be consuming all errors and then attempting to perform an API call to Discord. This will also perhaps fail since this coroutine requires an `int` type (which represents a User's ID) to be passed, yet you can potentially pass whatever `user` may be, or `str(ctx.author)`. 

However fetch_X methods all return status codes so I believe there is more at fault here than just the above. Perhaps you should join and ask in the [official help server](https://discord.gg/dpy).
